### ISSUE TITLE
Unable to download remote resources (such as user avatars) in the proxy network.

### ISSUE THREAD

**Issue by heimoshuiyu** [syuilo/Misskey#12886]
### 💡 Summary

I am in a special local area network.  All DNS requests within this LAN will be resolved to a special internal IP address. 

Then, the proxy request will be completed by this IP address. 

However, when Misskey recognizes that this address is a private address, it will refuse to download. 

I looked at the code and found that this check is made by this part of the code. 
https://github.com/misskey-dev/misskey/blob/4893cce43c97cf09c5f057fbbd302a8287ca984e/packages/backend/src/core/DownloadService.ts#L73C1-L78C5

```
}).on('response', (res: Got.Response) => {
	if ((process.env.NODE_ENV === 'production' || process.env.NODE_ENV === 'test') && !this.config.proxy && res.ip) {
		if (this.isPrivateIp(res.ip)) {
			this.logger.warn(`Blocked address: ${res.ip}`);
			req.destroy();
		}
	}
```

I wonder if we can remove these check, or, add an option in the configuration file to disable this check.? I am a Misskey novice, so if I have made a mistake, please let me know. Thank you.

### 🥰 Expected Behavior

Misskey will attempt to download regardless of the remote IP address.

### 🤬 Actual Behavior

Misskey recognizes the remote IP address as a private address and will refuse to download.

### 📝 Steps to Reproduce

Modify the /etc/hosts file to point a domain name to an internal IP address. Then attempt to use the Misskey backend to download resources from this domain. An error will occur.

### 💻 Frontend Environment

```markdown
* Model and OS of the device(s): -
* Browser: -
* Server URL: -
* Misskey: -
```

### 🛰 Backend Environment (for server admin)

```markdown
* Installation Method or Hosting Service: docker
* Misskey: docker 2023.12.2
* Node: -
* PostgreSQL: -
* Redis: -
* OS and Architecture: Linux
```

### Do you want to address this bug yourself?

- [X] Yes, I will patch the bug myself and send a pull request

**Comment by acid-chicken** [comment]
https://github.com/misskey-dev/misskey/issues/9220#issuecomment-1324392566

**Comment by heimoshuiyu** [comment]
I noticed that the preview of URLs in posts still did not work after changing the above configs. After some investigation, I found that it was necessary to set `SUMMALY_ALLOW_PRIVATE_IP` environment variable to `true`. I recorded this here so that anyone who encounters a similar issue can find it.

https://github.com/misskey-dev/summaly/blob/d2d8db49943ccb201c1b1b283e9d0a630519fac7/src/utils/got.ts#L46
